compiler/modules/types/list.bal

```diff
@@ -265,7 +273,12 @@ function listInhabited(Context cx, FixedLengthArray members, SemType rest, ListC
             if listInhabited(cx, members, NEVER, neg.next) {
                 return true;
             }
-            foreach int i in len + 1 ..< negLen {
+            // If last member of neg repeats, just checking one repeating occurrence should suffice.
```

Comment:
I don't understand why are not considering neg.maxInitialLen here.

Even if nt is the same after negLenLimit, the rest of the list may vary on each index up to neg.maxInitialLen.